src/main/java/com/timebook/timebook/controllers/UserController.java

```diff
@@ -23,14 +27,42 @@ public UserController(UserService userService) {
 
     // Add subscribed
     @PostMapping(value = "v1/subscribe")
-    public void addSubscription(@RequestBody String subscribeToEmail, Authentication authentication) {
-        UserData userInfo = (UserData) authentication.getPrincipal();
-        userService.createSubscription(subscribeToEmail, userInfo.getEmail());
+    public ResponseEntity<String> addSubscription(@RequestBody String subscribeToEmail, Authentication authentication) {
+        try {
+            UserData userInfo = (UserData) authentication.getPrincipal();
+            userService.createSubscription(subscribeToEmail, userInfo.getEmail());
+            return ResponseEntity.status(HttpStatus.SC_ACCEPTED).body("Successfully Add Subscription.");
+        } catch (UsernameNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.SC_BAD_REQUEST).body("Expectation Failed from Client (CODE 400)\n");
+        } catch (NullPointerException e) {
+            return ResponseEntity.status(HttpStatus.SC_BAD_REQUEST).body(e.getMessage());
+        }
     }
 
     @DeleteMapping(value = "v1/unsubscribe")
-    public void deleteSubscription(@RequestBody String unSubscribeToEmail, Authentication authentication) {
+    public ResponseEntity<String> deleteSubscription(@RequestBody String unSubscribeToEmail,
+            Authentication authentication) {
+        try {
+            UserData userInfo = (UserData) authentication.getPrincipal();
+            userService.deleteSubscription(unSubscribeToEmail, userInfo.getEmail());
+            return ResponseEntity.status(HttpStatus.SC_ACCEPTED).body("Successfully Delete Subscription.");
+        } catch (UsernameNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.SC_BAD_REQUEST).body("Expectation Failed from Client (CODE 400)\n");
+        } catch (NullPointerException e) {
+            return ResponseEntity.status(HttpStatus.SC_BAD_REQUEST).body(e.getMessage());
+        }
+    }
+
+    @PostMapping(value = "v1/lastview")
```

Comment:
Should this be an individual endpoint or just a service call when users retrieve data?